miio/tests/test_airhumidifier.py

```diff
@@ -467,6 +467,25 @@ def dry():
         self.device.set_dry(False)
         assert dry() is False
 
+    def test_water_level(self):
```

Comment:
I think it makes sense to use [pytest.parametrize](https://docs.pytest.org/en/6.2.x/example/parametrize.html) here:
```suggestion
    water_level_test_data = [(-1, 0), (0, 0), (60, 50), ...]
    @pytest.parametrize("depth, excepted", water_level_test_data)
    def test_water_level(self, depth, excepted):
        self.device.state["depth"] = depth
        assert self.state().water_level == expected
```

Reply:
Hmmm I think parametrization doesn't work with `unitteset`. Or I don't know how to use it properly.
```
self = <unittest.case._Outcome object at 0x7fe3287c4430>, test_case = <miio.tests.test_airhumidifier_miot.TestAirHumidifier testMethod=test_water_level>, isTest = True

    @contextlib.contextmanager
    def testPartExecutor(self, test_case, isTest=False):
        old_success = self.success
        self.success = True
        try:
>           yield

self = <miio.tests.test_airhumidifier_miot.TestAirHumidifier testMethod=test_water_level>
method = <bound method TestAirHumidifier.test_water_level of <miio.tests.test_airhumidifier_miot.TestAirHumidifier testMethod=test_water_level>>

    def _callTestMethod(self, method):
>       method()
E       TypeError: test_water_level() missing 2 required positional arguments: 'water_level' and 'expected'
```

Reply:
How does the code look like at the moment? Something like this?
```
@pytest.mark.parametrize("water_level,expected", ...)
def test_water_level(test, water_level, expected)
```

For some reason it doesn't seem to be passing the parameters in on that code path.

Reply:
Ah, I see, that's not possible when using `TestCase` classes: https://stackoverflow.com/questions/35561843/py-test-parametrize-test-cases-from-classes - maybe it's time to convert it to use regular test methods to avoid the whole class-based structure. Mind if I take a shot at it?